src/browser/convert_options.js

```diff
@@ -265,14 +265,18 @@ module.exports = {
         }
 
         const preload = options.preload;
-        if (!preload) {
-            // for all falsy values
-            newOptions.preload = [];
-        } else if (typeof preload === 'string') {
-            // backward compatibility
-            newOptions.preload = [{ url: preload }];
-        } else {
-            newOptions.preload = preload;
+        if (preload) {
+            if (typeof preload === 'string') {
+                // convert legacy `preload` option into modern `preload` option
+                newOptions.preload = [{ url: preload }];
+            } else if (
+                Array.isArray(preload) &&
+                preload.every(eachPreload => typeof eachPreload === 'object' && typeof eachPreload.url === 'string')
+            ) {
+                newOptions.preload = preload;
+            } else {
+                log.writeToLog('warning', 'Expected `preload` option to be a string primitive OR an array of objects with `url` string properties.');
```

Comment:
Lets not fix this right now because its out of scope but we should error out if the preload script format is wrong to the point we cannot take any actions, never fail silently.

Reply:
https://appoji.jira.com/browse/RUN-3371